pkg/resource/imageregistryca.go

```diff
@@ -0,0 +1,187 @@
+
+type generatorImageRegistryCA struct {
+	lister                    corelisters.ConfigMapLister
+	imageConfigLister         configlisters.ImageLister
+	openshiftConfigLister     corelisters.ConfigMapNamespaceLister
+	serviceLister             corelisters.ServiceNamespaceLister
+	imageRegistryConfigLister imageregistryv1listers.ConfigLister
+	storageListers            *client.StorageListers
+	kubeconfig                *restclient.Config
+	client                    coreset.CoreV1Interface
+}
+
+func NewGeneratorImageRegistryCA(
+	lister corelisters.ConfigMapLister,
+	imageConfigLister configlisters.ImageLister,
+	openshiftConfigLister corelisters.ConfigMapNamespaceLister,
+	serviceLister corelisters.ServiceNamespaceLister,
+	imageRegistryConfigLister imageregistryv1listers.ConfigLister,
+	storageListers *client.StorageListers,
+	kubeconfig *restclient.Config,
+	client coreset.CoreV1Interface,
+) Mutator {
+	return &generatorImageRegistryCA{
+		lister:                    lister,
+		imageConfigLister:         imageConfigLister,
+		openshiftConfigLister:     openshiftConfigLister,
+		serviceLister:             serviceLister,
+		imageRegistryConfigLister: imageRegistryConfigLister,
+		storageListers:            storageListers,
+		kubeconfig:                kubeconfig,
+		client:                    client,
+	}
+}
+
+func (girca *generatorImageRegistryCA) Type() runtime.Object {
+	return &corev1.ConfigMap{}
+}
+
+func (girca *generatorImageRegistryCA) GetNamespace() string {
+	return defaults.OpenShiftConfigManagedNamespace
+}
+
+func (girca *generatorImageRegistryCA) GetName() string {
+	return defaults.ImageRegistryCAName
+}
+
+func (girca *generatorImageRegistryCA) storageDriver() (storage.Driver, bool, error) {
+	imageRegistryConfig, err := girca.imageRegistryConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		return nil, false, nil
+	} else if err != nil {
+		return nil, false, err
+	}
+
+	if imageRegistryConfig.Spec.ManagementState == operatorv1.Removed {
+		// The certificates controller does not need to know about
+		// storage when the management state is Removed.
+		return nil, false, nil
+	}
+
+	driver, err := storage.NewDriver(&imageRegistryConfig.Spec.Storage, girca.kubeconfig, girca.storageListers)
+	if err == storage.ErrStorageNotConfigured || storage.IsMultiStoragesError(err) {
+		return nil, false, nil
+	} else if err != nil {
+		return nil, false, err
+	}
+
+	canRedirect := !imageRegistryConfig.Spec.DisableRedirect
+
+	return driver, canRedirect, nil
+}
+
+func (girca *generatorImageRegistryCA) expected() (runtime.Object, error) {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      girca.GetName(),
+			Namespace: girca.GetNamespace(),
+		},
+		Data:       map[string]string{},
+		BinaryData: map[string][]byte{},
+	}
+
+	var ownHostnameKeys []string
+
+	serviceCA, err := girca.lister.ConfigMaps(defaults.OpenShiftConfigManagedNamespace).Get(defaults.ServiceCAName)
```

Comment:
`serviceca` is managed by the image-registry operator and its namespace is `openshift-image-registry`.

The generator is https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/resource/serviceca.go